core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -80,14 +110,55 @@ class ControllerContext {
     replicaStates.clear()
   }
 
+  def partitionReplicaAssignment(topicPartition: TopicPartition): Seq[Int] = {
+    partitionAssignments.getOrElse(topicPartition.topic, mutable.Map.empty)
+      .get(topicPartition.partition) match {
+        case Some(partitionAssignment) => partitionAssignment.replicas
+        case None => Seq.empty
+      }
+  }
+
+  def partitionFullReplicaAssignment(topicPartition: TopicPartition): PartitionReplicaAssignment = {
+    partitionAssignments.getOrElse(topicPartition.topic, mutable.Map.empty)
+      .get(topicPartition.partition) match {
+      case Some(partitionAssignment) => partitionAssignment
+      case None => PartitionReplicaAssignment(Seq(), Seq(), Seq())
```

Comment:
Is there any benefit to this being `PartitionReplicaAssignment.EMPTY`? It's probably easier to read